src/Ogooreck.Sample.BusinessLogic.FSharp.Tests/Deciders/BankAccountTests.fs

```diff
@@ -0,0 +1,109 @@
+module BankAccountTests
+
+open System
+open Deciders.BankAccount
+open Ogooreck.BusinessLogic
+open Deciders.BankAccountDecider
+open Xunit
+
+
+let random = Random()
+let now = DateTimeOffset.UtcNow
+
+let getNow = fun () -> now
+
+let spec =
+    Specification.For(decide getNow, evolve)
+
+let BankAccountOpenedWith =
+    fun bankAccountId now version ->
+        let accountNumber = Guid.NewGuid().ToString()
+
+        let clientId = Guid.NewGuid()
+        let currencyISOCode = "USD"
+
+        { BankAccountId = bankAccountId
+          AccountNumber = accountNumber
+          ClientId = clientId
+          CurrencyISOCode = currencyISOCode
+          CreatedAt = now
+          Version = version }
+        |> Event.BankAccountOpened
+
+let BankAccountClosedWith =
+    fun bankAccountId now version ->
+        { BankAccountId = bankAccountId
+          Reason = Guid.NewGuid().ToString()
+          ClosedAt = now
+          Version = version }
+        |> Event.BankAccountClosed
+
+[<Fact>]
+let ``GIVEN non existing bank account WHEN open with valid params THEN bank account is opened`` () =
+    let bankAccountId = Guid.NewGuid()
+
+    let accountNumber =
+        Guid.NewGuid().ToString()
+
+    let clientId = Guid.NewGuid()
+    let currencyISOCode = "USD"
+
+    spec
+        .Given()
```

Comment:
How to represent non-existing bank account so that it matches the test case description?

Reply:
It's a stream with no events. I could put here empty array if that helps 🤔

Reply:
Initially it might look a bit weird to see such a silly construct:
```fsharp
let nonExistingBankAccount = List.empty<_>
```

But the meaning will be conjured in:
```fsharp
spec
	.Given(nonExistingBankAccount)
//...
```

It gives the same context as in the test case description.

Reply:
Makes sense to me 👍 I applied the suggestion in https://github.com/oskardudycz/Ogooreck/pull/13/commits/95889d7da2d5e8124248a23e0fa97b97b6b3d5bc.